openid-data-classes/src/commonMain/kotlin/at/asitplus/openid/RequestParametersFrom.kt

```diff
@@ -36,16 +73,36 @@ sealed class RequestParametersFrom<S : RequestParameters> {
     }
 
     @Serializable
+    @SerialName(SerialNames.TYPE_JSON)
     data class Json<T : RequestParameters>(
+        @SerialName(SerialNames.JSON_STRING)
         val jsonString: String,
+        @SerialName(SerialNames.PARAMETERS)
         override val parameters: T,
-        val dcApiRequest: DCAPIRequest? = null
+        @SerialName(SerialNames.PARENT)
+        val parent: Url?,
     ) : RequestParametersFrom<T>() {
         override fun toString(): String {
-            return "Json(jsonString='$jsonString', parameters=$parameters)"
+            return "Json(parent='$parent', jsonString='$jsonString', parameters=$parameters)"
         }
     }
+
+    object SerialNames {
+        const val TYPE_JWS_SIGNED = "JwsSigned"
+        const val TYPE_JSON = "Json"
+        const val TYPE_DCAPI_UNSIGNED = "DcApiUnsigned"
+        const val TYPE_DCAPI_SIGNED = "DcApiSigned"
+        const val TYPE_URI = "Uri"
```

Comment:
No need to change that if you feel it is overkill, but you could add another layer of nesting to the constants so that type constants and property constants are separated (`TYPE_JWS_SIGNED` and `JWS_SIGNED` in the same object has the most footgun potential)